copy-n-paste (#1)
Kosuke-iga/SSS-IEEEAccess: access.tex

However, that scheme can be evaded by the adversary with manipulated malware grayscale image or binary data.
In this thesis, we propose an evasion-resilient IoT malware detection scheme with invalidating adversarial byte sequences and 1D convolutional filters.
The valuable regions still remain which contribute to classification in both manipulated targets, grayscale image and binary data, after the attacks.
Thus, I improve the detection accuracy by statically extracting/enhancing valuable region of each manipulated target to the CNN model.
By evaluation with each manipulated dataset, we show our scheme can improve detection performance against both evasive techniques. 
Exploring the valuable regions remaining in manipulated target and improvement of the detection accuracy utilizing the regions in my detection are the contributions of my research. 
\end{abstract}

\begin{keywords} 
 IoT malware detection, Convolutional Neural Network, Obfuscation, Adversarial Examples 
 \end{keywords}

\titlepgskip=-15pt

\maketitle

\section{Introduction}  \label{sec:introduction}

> In these days, the Internet of Things (IoT) is interconnetcting a large number of electronic devices with a variety of applications in our lives, such as smart appliances, smart houses, smart grid, energy management systems, and so on, at a tremendous speed.

Reviewer: ```suggestion
In these days, the Internet of Things (IoT) is interconnetcting a large number of electronic devices with a variety of applications in our lives, such as smart appliances, smart houses, smart grid, energy management systems at a tremendous speed.

```

The number of the IoT devices is continuously increased.
It is estimated to be about 50 billion all over the world by 2030 \cite{bg1, bg2}.
They become targets of malware attacks due to the rapid increase and the development into devices which have interconnectivity in these days.

> Unfortunately, vulnerable IoT devices are spreading since the countermeasures cannot keep up with this trend of IoT malware attacks.

Reviewer: ```suggestion
Unfortunately, vulnerable IoT devices are spreading since the countermeasures cannot keep up with the evolution of IoT malware.

```

\figurename~\ref{fig:mirai} represents an actual case of attack damage by malware called Mirai.

> In fact, as shown in \figurename~\ref{fig:mirai}, attacks against companies using many IoT devices infected with malware have become a problem, and the detection of IoT malware has become an urgent issue.

Reviewer: ```suggestion
In fact, as shown in \figurename~\ref{fig:mirai}, attacks against companies using many IoT devices infected with malware have become a problem.
```

Thus, this circumstance results in the urgency of detecting IoT malware.

\begin{figure*}[t]
 \centering
 %\hspace{-55pt}
 \includegraphics[scale=0.56]{./figures/mirai.pdf}
 \caption{The case of attack damage by Mirai malware.} 
 \label{fig:mirai}
\end{figure*}
%\afterpage{\clearpage}
%\newpage
Existing solutions for detecting malware are mainly classified into router-side schemes \cite{net} and device-side schemes \cite{om, hwang, cfg, cfg2}.

Router-side schemes pay attention to the fact that there exist strong evidences of malware in network traffic.

> This is because adversaries have to scan IoT devices and propagate their malware before attacking itself and these phases cause abnormal traffic which cannot be seen in benign case.

Reviewer: ```suggestion
This is because adversaries have to scan IoT devices and propagate their malware before attacks itself.
As a result, these phases cause abnormal traffic which cannot be seen in benign cases.

```

> Based on the fact, those scheme utilize the features extracted from incoming packets in the scanning/propagation phase such as destination IP address during a certain time, port number and so on, at a gateway of the network.

Reviewer: and so on は冗長かな
```suggestion
Based on the fact, those schemes utilize the features such as destination IP address during a certain time, port number, at a gateway of the network.
The features are extracted from incoming packets in the scanning/propagation phase.
```

> Although router-side schemes are useful, these schemes are subject to internal attacks of the local network itself since the packets do not go through the router.

Reviewer: since は既知の情報を言う時に使って、becauseは未知の情報を言う時に使う様にしたい。
```suggestion
Although router-side schemes are useful, these schemes are subject to internal attacks of the local network itself because the packets do not go through the router.
```

In order to cope with the limitations of router-side schemes, device-side schemes are proposed.
Device-side schemes are conducted by utilizing the features extracted from malware itself at each IoT devices.

> Most of the device-side schemes utilze various features which can be extracted statically since IoT devices are not suitable for dynamic detections due to their limited computing and storage capabilities.

Reviewer: ```suggestion
Most of the device-side schemes utilze various features which can be extracted statically because IoT devices are not suitable for dynamic detections due to their limited computing and storage capabilities.
```

> Kumar et al. utilezes the features regarding to Control Flow Graphs (CFGs) acquired by complicated analysis and definitions of the features \cite{cfg}.

Reviewer: ```suggestion
Kumar et al. utilezes the features regarding Control Flow Graphs (CFGs) acquired by complicated analysis and definitions of the features \cite{cfg}.
```

Although that scheme can extract represantative features of the application, it is a time-consuming and laborious method due to the analysis for an enormous number of IoT malware.

In order to cope with the spreading IoT malware with device-side detections, some researchers utilize Convolutional Neural Network (CNN) in their scheme.
Su et al. utilezes GrayScale Image (GSI) converted from raw bytes of each malware binary for its CNN \cite{previous}.

> Although it is suitable for IoT malware detection since CNN can extract malware features from GSI automatically without feature engineering, it can be avoided with evasive techniques by adversaries to GSI or Binary Data (BD) of malware. 

Reviewer: このへんでAMとOMは話しておくべき。
問題点の方には書いてあるから、それをうまくイントロでまとめて反映する感じだな。
あと、1文が長いので、分けた。
とりあえずこんな感じ。
```suggestion
That scheme is suitable for IoT malware detection because CNN can extract useful features from GSI automatically without feature engineering.
However, it can be avoided with two representative evasive techniques, namely adversarial example and obfuscation by adversaries.
As a result, Adversarial Malware (AM) and Obfuscated Malware (OM) are created by applying these techniques to GSI or Binary Data (BD) of malware, respectively. 
```

In order to detect such malware with exisiting evasive techniques, that is manipulated GSIs and BD, in this thesis, we propose an evasion-resilient IoT malware detection scheme with invalidating adversarial byte sequences and 1D convolutional filters.
The main idea of our scheme is that the some regions which contribute to decision making in each manipulated target still remain after the evasive techniques by adverasaries. 
This research aims to improve the detection accuracy by statically by statically extracting/enhancing each beneficial region of the manipulated target for the CNN model.

The contributions of this thesis are as follows: 
\begin{itemize}

>  \item I find the valuable regions which remain in adversarial malware and improve the detection accuracy of them by denoising procedure.

Reviewer: ```suggestion
 \item we found the valuable regions which remain in adversarial malware and improve the detection accuracy of them by denoising procedure.

```

>  \item I find the valuable features which obfuscated malware have and improve the detection accuracy of them by emphasizeing their valuable regions to CNN model.

Reviewer: we found

>  \item To the best of my knowledge, there is currenlty no reference to research defense schemes against the evasive techniques by adversaries. However, it is considered in this thesis, and, furthermore, give effective solutions.

Reviewer: 具体的にAMと難読化の両方に対応するっていた方がインパクトあり。
```suggestion
 \item To the best of my knowledge, there is currenlty no reference to research defense schemes against the evasive techniques, namely AM generation and obfuscation by adversaries. 
Thus, we propose effective solutions that counter both of them.
```

\end{itemize}

The rest of this thesis is constructed as follows. 
After this introduction, related work are introduced in Section \ref{sec:related_work}.
The previous scheme and the issues of that scheme are explained in Section \ref{sec:previous_scheme}.
Proposed scheme is described in Section \ref{sec:proposed_scheme}.
Various evaluation results are shown in Section \ref{sec:evaluation}.
Finally, the conclusions of this thesis are presented in Section \ref{sec:conclusion}.

\section{Related Work} \label{sec:related_work}
In order to detect IoT malware, several schemes have been proposed. 
These schemes are divided into router-side detection and device-side detection.
Router-side detection is performed at a gateway on the basis of the features regarding propagation behavior for building a botnet.
Meanwhile, device-side detection is conducted at each IoT device by utilizing static features extracted from malware itself.
The representative schemes are explained in the following subsections.

\subsection{Router-side detection}
Router-side detctions leverage the fact that there exists strong evidence of malware in network traffic.

> Kumar et al. \cite{net} propose the scheme focusing on network traffic while the scanning and propagation phase in particular which is observed in most of the existing malware behaviour for the purpose of building a botnet as shown in \figurename~\ref{fig:lan}.

Reviewer: ```suggestion
Kumar et al. \cite{net} propose the scheme focusing on network traffic while the scanning and propagation phase. 
Most of the existing malware behaviour in the phase are carried out in order to build a botnet as shown in \figurename~\ref{fig:lan}.
```

> Bots scanning for and infecting vulnerable devices are targeted in particular that scheme. 

Reviewer: ```suggestion
Therefore, that scheme aims at detecting the bots scanning and infecting vulnerable devices before they paticipate in a botnet for an actural attack such as DDoS.
```

> This is because the scanning and propagation phase of the botnet life-cycle stretches over many months and that scheme can detect and isolate the bots before they can participate in an actual attack such as DDoS.

Reviewer: ```suggestion
That scheme can detect and isolate the bots before attacks because the scanning and propagation phase of the botnet life-cycle stretches over many months, which results in abnormal traffic that cannot be seen in benign cases.
```

> Based on the fact above, that scheme detects abnormal traffic caused while this phase which cannot be seen in benign cases utilizing machine learning.

Reviewer: ```suggestion
The abnormal traffic caused while this phase can be detected by using machine learning.
```

> It utilizes the number of unique destination IP addresses and packets per destination IP addresses as features observing incoming packets which have spcific target destination port number that the existing malware tend to send.

Reviewer: ```suggestion
It utilizes the number of unique destination IP addresses and packets per destination IP addresses as features observing incoming packets which have spcific target destination port number that the existing malware tends to send.
```

> The number of unique destination IP addresses in case of malware-induced scanning traffic will be far more than benign traffic since the malware generates random IP addresses and sends malicious requests to them.

Reviewer: ながくね？？w
一つの方式を詳しく述べすぎな気がする。ここより上くらいの説明で十分じゃないかな？
他にルータサイドの検知手法ないの？？
無駄に長く書くくらいなら2つまとめた方が良い。

Also, the number of packets per destination IP addresses seeks to exploit the fact that malware typically does not send mulitiple malicious packets to the same IP addresses (only a single packet is sent in most cases), possibly to cover as many devices as possible during the scanning/propagation phase.

> By being based on the scanning/propagation phase, it can detect and isolate the bots before they can participate in an actual attack such as DDoS.

Reviewer: もう上で言ってるので、削除
```suggestion
```

However, that scheme is subject to local network attacks where IoT devices are targeted directly by adversaries in the same local network.
Thus, exploring the malware detection schemes which can be realized by each IoT device is needed due to the vulnerability against local network attacks.

\begin{figure}[t]
 \centering
 \includegraphics[scale=0.3]{./figures/lan.pdf}
 \caption{The router-side detection in the scanning/propagation flow and its problem.} 
 \label{fig:lan}
\end{figure}

\subsection{Device-side detection} 

> Device-side detection schemes are conducted by utilizing the features extracted from malware file itself at each device on the basis of malware, and they tend to have similality since adversaries make them based on the existing ones \cite{om, hwang, cfg, cfg2}.

Reviewer: ```suggestion
Device-side detection schemes are conducted by utilizing the features extracted from a malware file itself at each device.
They tend to have similality because adversaries create new malware based on the existing ones \cite{om, hwang, cfg, cfg2} in many cases.
```

Most of them utilize various features which can be extracted statically since IoT devices are not suitable for dynamic approach due to their limited computing and storage capabilities.

Torabi et al. propose the scheme which utilizes meaningful strings from the binary code, such as IP addresses of adversarial hosts (e.g., C\&C servers) and/or embedded commands/payloads \cite{om}.

> That scheme focus on the fact that IoT malware need to communicate with adversarial hosts to obtain malicious command/payload and upload gathered information.

Reviewer: malware は不可算名詞らしいよ。
```suggestion
That scheme focuses on the fact that IoT malware needs to communicate with adversarial hosts to obtain malicious command/payload and upload gathered information.
```

> This is typically achieved by embedding a series of commands and IP addresses to ensure successful post-infection communication for operating further malicious activities. 

Reviewer: ```suggestion
This is typically achieved by embedding a series of commands and IP addresses to ensure successful post-infection communication for conducting further malicious activities. 
```

> Although that scheme realizes lightweight static detection with simple reverse-engineering techniques, its detection can be easily avoided by malware where the meaningful strings are obfuscated in order to hide them with simple obfuscation techniques.

Reviewer: シンプルな難読化技術で隠すために意味のある文字列を難読化するマルウェアって小泉進次郎みたいになってるので、しっかり意味を考えて書こう。
他にもそういう箇所が見られる。
```suggestion
Although that scheme realizes lightweight static detection with simple reverse-engineering techniques, its detection can be easily avoided by malware where the meaningful strings are obfuscated in order to prevent them from being extracted.
```

Such obfuscation techniques can be applied even on IoT devices where the computing and storage capabilities are limited.

> Hwang et al. also use string information from binary data \cite{hwang}.

Reviewer: これここにおくの変だと思う。
難読化されるからダメだって言ったあとにこれを紹介するのは文章の流れとしておかしいので、上の方式と並行して書く様にするべき。
できれば、ほとんど差がないから、違う方式ある方が流れとして綺麗だし関連研究のセクションっぽいんだけど…ね
```suggestion
Hwang et al. also use string information from binary data \cite{hwang}.
```

> They extract around 1800 different strings including API names, DLL names, library function names of programming languages and PE/ELF file formats.

Reviewer: DLLってなんの略？？
書いた方が良い。
```suggestion
They extract around 1,800 different strings including API names, DLL names, library function names of programming languages and PE/ELF file formats.
```

> The major limitation of their work is also that their method may not be effective over packed or encrypted samples because their method depends only on the strings available explicitly, and such packed/encrypted samples reveal limited string information during static analysis.

Reviewer: 修論のかさ増し感が否めないw
同じこと言ってるので、冗長。
```suggestion
The major limitation of their work is also that their method may not be effective over packed or encrypted samples because their method depends only on the strings available explicitly, and such packed/encrypted samples reveal limited string information during static analysis.
```

> In order to avoid easily detection rejection by adversaries, researchers propose the scheme which utilizes the features acquired from a Control Flow Graph (CFG), which could be used to extract representative static features of the application as shown in \figurename~\ref{fig:cfg} \cite{cfg, cfg2}.

Reviewer: あと、FIGUREじゃなくて、Fig.にしないといけなかったと思う。
俺のお古のtex使ってるはずだから、myfigurenameみたいなので定義しているはずなので、他もそっちに変更しておいて。
```suggestion
In order to avoid easily detection rejection by adversaries, researchers \cite{cfg, cfg2} propose the scheme which utilizes the features acquired from a Control Flow Graph (CFG), which could be used to extract representative static features of the application.
\figurename~\ref{fig:cfg} shows the overview of device-side detection using CFG.
```

> The theoretic metrics of CFGs, such as the number of edges, density of a graph, shortest paths between each node and so on, can be multiple features of the model effctively derived from malware constructions under the limitation of static approach.

Reviewer: 他もそうだけど、such as の場合は、and so on は冗長なのでいらないと思う。
```suggestion
The theoretic metrics of CFGs are the number of edges, density of a graph, shortest paths between each node and so on.
They can be multiple features of a detection model and be derived from malware constructions under the limitation of static approaches.
```

Alasmary et al. focus on the fact that some defferences in those metrics can be observed between benign software and IoT malware \cite{cfg}.

> In that research, they reveal that IoT malware, and also each malware family, have various algorithmic and structural properties through complex analysis of CFGs acquired from about 6000 malware and benign software as shown in \figurename~\ref{fig:cfg}. 

Reviewer: ```suggestion
In that research, they reveal that IoT malware, and also each malware family, have various algorithmic and structural properties through complex analysis of CFGs acquired from about 6,000 malware and benign software as shown in \figurename~\ref{fig:cfg}. 
```

> Utilizing such properties of the CFGs, that scheme realize high accurate detection.

Reviewer: ```suggestion
Utilizing such properties of the CFGs, that scheme realizes highly accurate detection.

```

> However, that scheme is a time-consuming and laborious method due to the feature engineering for exploring effective metrics with analysing an enormous number of IoT malware.

Reviewer: ここの下に従来の話もちょこっと書いて、これに着目するよって書いてから、previous scheme にいくべき。
突然のprevious scheme は査読者に疑問を生ませてしまう。
特にアクセスは1週間しか査読期間がないから、疑問をなるべく生じさせない様に書くことが何よりも重要。

\begin{figure}[t]
 \centering
 \includegraphics[scale=0.3]{./figures/cfg.pdf}
 \caption{The overview of device-side detection using CFG.} 
 \label{fig:cfg}
\end{figure}

\section{Previous scheme} \label{sec:previous_scheme}
\subsection{Overview} 
In order to deal with spreading IoT malware with device-side detections in more practical means, CNN is utilized in the previous scheme \cite{previous}.

> Our research is more practical scheme which applys this device-side detection with CNN improving its issues.

Reviewer: ここは、従来の話だから、俺らの取り組みの話はいらないかな。
単純にアイデアとやったことを書くと良いかと。
これ書くならイントロかなって感じ。
```suggestion
Our research is more practical scheme which applys this device-side detection with CNN improving its issues.
```

Author: ちなみにこれは寺岡さんに入れろって言われましたねｗ

The main idea of the previous scheme is that there exists the difference of GSIs converted from each malware Binary Data (BD) which is always acquired in IoT devices due to the exectable format of them.

> In that scheme \cite{previous}, a malware binary is reformatted as an 8-bit sequence and then be converted to a GSI which has one channel and pixel values from 0 to 255.

Reviewer: a bynary code じゃない？？
```suggestion
In the previous scheme \cite{previous}, a malware binary code is reformatted as an 8-bit sequence and then is converted to a GSI which has one channel and pixel values from 0 to 255.

```

> To confirm the performance of the scheme, some samples of malware and benign software GSIs are prepared in our research and shown by \figurename~\ref{fig:malGSI} and \figurename~\ref{fig:benGSI}.

Reviewer: 性能評価始まるわけではないので、性能を確かめるとか言わなくていいのでは？
遠回しな気がする。
```suggestion
To confirm the differences of malware and benign software, we prepared some samples of malware and benign software GSIs and compared them.
\figurename~\ref{fig:malGSI} and \figurename~\ref{fig:benGSI} show the GSIs of malware and the ones of benign software, respectively.
```

By comparison, some novel differences between them can be observed even by human eyes.
For example, it can be seen that malware GSIs always are more dense.
On the other hand, the GSI of benign softwares tends to have larger header parts than malware.

> These GSIs can represents each features and be fed into CNN.

Reviewer: ```suggestion
These GSIs can represents each features, which means useful features.
Fortunately, CNN can extract effective features from these GSI automatically by learning deep non-linear features that can be hardly descovered and understood even by human eyes.
```

> Since, CNN can extract effective features from these GSI automatically by learning deep non-linear features even that can be hardly descovered and understood by human eyes.

Reviewer: Since, ？？
```suggestion
```

> Furthermore, once trained, the network itself is lightweight and can be run with tiny computational resources, since only the trained parameters and information of network structure are kept.

Reviewer: ```suggestion
Furthermore, once trained, the network model itself can be run with tiny computational resources, because only the trained parameters and information of network structure are shared, which results in lightweight detection on IoT devices.
```

Thus, CNN is a suitable scheme for IoT malware detections since it does not demand feature engineering and also computational costs while the detection.

\begin{figure}[t]
 \centering
 %\hspace{-55pt}
 \includegraphics[scale=0.35]{./figures/malGSI.pdf}

>  \caption{The malware GSIs.} 

Reviewer: ```suggestion
 \caption{The GSIs of malware.} 
```

 \label{fig:malGSI}
\end{figure}
\begin{figure}[t]
 \centering
 \includegraphics[scale=0.3]{./figures/benGSI.pdf}

>  \caption{The benign file GSIs.} 

Reviewer: file か softwareで統一した方が良い。
わかりにくい。
```suggestion
 \caption{The GSIs of benign software.} 
```

 \label{fig:benGSI}
\end{figure}

> \subsection{Issues} 

Reviewer: IEEEの論文の書き方みたいなやつで、Issueは使わない様にって書かれてあった気がする。
shortcoming か Problem がよいかな。
```suggestion
\subsection{Shortcomings of the Previous Scheme} 
```
